app/src/main/java/org/simple/clinic/teleconsultlog/success/TeleConsultSuccessScreen.kt

```diff
@@ -0,0 +1,104 @@
+package org.simple.clinic.teleconsultlog.success
+
+import android.content.Context
+import android.os.Parcelable
+import android.util.AttributeSet
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.cast
+import kotlinx.android.synthetic.main.screen_teleconsult_success.view.*
+import org.simple.clinic.di.injector
+import org.simple.clinic.home.HomeScreenKey
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.router.screen.RouterDirection.BACKWARD
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.util.unsafeLazy
+import javax.inject.Inject
+
+class TeleConsultSuccessScreen(
+    context: Context,
+    attributeSet: AttributeSet
+) : ConstraintLayout(context, attributeSet), TeleConsultSuccessScreenUiActions {
+
+  @Inject
+  lateinit var effectHandler: TeleConsultSuccessEffectHandler.Factory
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  private val events: Observable<TeleConsultSuccessEvent> by unsafeLazy {
+    Observable
+        .merge(
+            backClicks(),
+            yesClicks(),
+            noClicks()
+        )
+        .cast<TeleConsultSuccessEvent>()
```

Comment:
I think we can skip adding the explicit `<TeleConsultSuccessEvent>` to the `.cast` operator since Kotlin's type inference is smart enough to figure it out.